cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+	} else {
+		log.Entry().Infof("Manifest file '%s' was not in legacy format. No tranformation needed, no update performed.", manifestFile)
+	}
+	return nil
+}
+
+func prepareCfPushCfNativeDeploy(config *cloudFoundryDeployOptions) (string, []string, []string, error) {
+
+	deployOptions := []string{}
+	varOptions, err := getVarOptions(config.ManifestVariables)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+	varFileOptions, err := getVarFileOptions(config.ManifestVariablesFiles)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+
+	deployOptions = append(deployOptions, varOptions...)
+	deployOptions = append(deployOptions, varFileOptions...)
+
+	return "push", deployOptions, []string{}, nil
+}
+
+func toStringInterfaceMap(in *orderedmap.OrderedMap, err error) (map[string]interface{}, error) {
+
+	out := map[string]interface{}{}
+
+	if err == nil {
+		for _, key := range in.Keys() {
+			if k, ok := key.(string); ok {
+				val, exists := in.Get(key)
+				if exists {
+					out[k] = val
+				} else {
+					return map[string]interface{}{}, fmt.Errorf("No entry found for '%v'", key)
+				}
+			} else {
+				return map[string]interface{}{}, fmt.Errorf("Cannot cast key '%v' to string", key)
+			}
+		}
+	}
+
+	return out, err
+}
+func getVarOptions(vars []string) ([]string, error) {
+
+	varsMap, err := toParameterMap(vars)
+	if err != nil {
+		return []string{}, err
+	}
+
+	varsResult := []string{}
+
+	for _, key := range varsMap.Keys() {
+		val, _ := varsMap.Get(key)
+		if v, ok := val.(string); ok {
+			varsResult = append(varsResult, "--var", fmt.Sprintf("%s=%s", key, quoteAndBashEscape(v)))
+		} else {
+			return []string{}, fmt.Errorf("Cannot cast '%v' to string", val)
+		}
+	}
+	return varsResult, nil
+}
+
+func getVarFileOptions(varFiles []string) ([]string, error) {
+
+	varFilesResult := []string{}
+
+	for _, varFile := range varFiles {
+		fExists, err := fileUtils.FileExists(varFile)
+		if err != nil {
+			return []string{}, err
+		}
+
+		if !fExists {
+			log.Entry().Warningf("We skip adding not-existing file '%s' as a vars-file to the cf create-service-push call", varFile)
+			continue
+		}
+
+		varFilesResult = append(varFilesResult, "--vars-file", quoteAndBashEscape(varFile))
+	}
+
+	if len(varFilesResult) > 0 {
+		log.Entry().Infof("We will add the following string to the cf push call: '%s'", strings.Join(varFilesResult, " "))
+	}
+	return varFilesResult, nil
+}
+
+func quoteAndBashEscape(s string) string {
+	escapedSingleQuote := "'\"'\"'"
+	return strings.ReplaceAll(s, "'", escapedSingleQuote)
+}
+
+func checkAndUpdateDeployTypeForNotSupportedManifest(config *cloudFoundryDeployOptions) (string, error) {
+
+	manifestFile := config.Manifest
+	if len(manifestFile) == 0 {
+		manifestFile = "manifest.yml"
+	}
+	var manifestFileExists bool
+	var err error
+
+	if len(manifestFile) > 0 {
+		manifestFileExists, err = fileUtils.FileExists(manifestFile)
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
```suggestion
	manifestFileExists, err := fileUtils.FileExists(manifestFile)
	if err != nil {
		return "", err
	}
```

Reply:
Does this compile on your end? Here on my side it doesn't compile.

The `manifestFileExists` is set inside a nested control flow. When we remove that the variable is not found at a higher level in the testing.

When we keep the `manifestFileExists` as it is but remove the `err` I get `manifestFileExists declared and not used`. This means the variable is re-declared inside the block. The assignment happens to the closer variable in this case resulting in the message wrt variable "above".

Please update me in case this is different on your end.

Reply:
The point is that due to lines 594-596, it is impossible that `len(manifestFile) > 0` will be `false`. My suggestion is to eliminate this check. Then there will be no more extra scoping. Just looking at the complete suggestion above, I am almost sure that if you apply it whole, then it should compile.

Reply:
got is -> simplfied